src/EditorFeatures/CSharpTest/ImplementInterface/ImplementExplicitlyTests.cs

```diff
@@ -562,5 +561,67 @@ int IBar.M(int i)
     }
 }", index: SingleMember);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        [WorkItem(52020, "https://github.com/dotnet/roslyn/issues/52020")]
+        public async Task TestWithContraints()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+interface IRepro
+{
+    void A<T>(int value) where T : class;
+}
+
+class Repro : IRepro
+{
+    public void [||]A<T>(int value) where T : class
+    {
+    }
+}",
+@"
+interface IRepro
+{
+    void A<T>(int value) where T : class;
+}
+
+class Repro : IRepro
+{
+    void IRepro.A<T>(int value)
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        [WorkItem(52020, "https://github.com/dotnet/roslyn/issues/52020")]
+        public async Task TestWithDefaultParameterValues()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+interface IRepro
+{
+    void A(int value = 0);
+}
+
+class Repro : IRepro
+{
+    public void [||]A(int value = 0)
```

Comment:
❗ Need to add a test where the default value in the implementation differs from the default value in the interface. For this case, the default value _should not_ be removed during the conversion. The user will need to make a conscious choice between going back to the implicit implementation (which will properly round-trip the non-inherited value), or accepting the new default.

Reply:
I was bitten by this back in 2014-2015 working on a different project. Right now, there is no analyzer support for the scenario (https://github.com/dotnet/roslyn-analyzers/issues/3639), so it's really difficult for a user to understand the semantics change from a seemingly-innocent refactoring.

Reply:
Awesome!